index.js

```diff
@@ -69,74 +87,96 @@ module.exports = {
+  },
+
+  _shouldCompileModules: function(addonOptions) {
+    var babelOptions = addonOptions.babel;
+    var customOptions = addonOptions['ember-cli-babel'];
+
+    if (customOptions && 'compileModules' in customOptions) {
+      return customOptions.compileModules === true;
+
+    } else if (babelOptions && 'compileModules' in babelOptions) {
+      if (this._shouldShowBabelDeprecations && !this._modulesDeprecationPrinted) {
+        this._modulesDeprecationPrinted = true;
+        this.ui.writeDeprecateLine(
+          'Putting the "compileModules" option in "babel" is deprecated, please put it in "ember-cli-babel" instead.');
       }
-    }
-  };
 
-  // Ensure modules aren't compiled unless explicitly set to compile
-  options.blacklist = options.blacklist || ['es6.modules'];
+      return babelOptions.compileModules === true;
+    } else {
+      return false;
+    }
+  },
 
-  // do not enable non-standard transforms
-  if (!('nonStandard' in options)) {
-    options.nonStandard = false;
-  }
+  _getBabelOptions: function() {
+    var addonOptions = this._getAddonOptions();
+    var options = clone(addonOptions.babel || {});
+
+    var compileModules = this._shouldCompileModules(addonOptions);
+
+    var ui = this.ui;
+
+    // pass a console object that wraps the addon's `UI` object
+    options.console = {
+      log: function(message) {
+        // fallback needed for support of ember-cli < 2.2.0
+        if (ui.writeInfoLine) {
+          ui.writeInfoLine(message);
+        } else {
+          ui.writeLine(message, 'INFO');
+        }
+      },
+
+      warn: function(message) {
+        // fallback needed for support of ember-cli < 2.2.0
+        if (ui.writeWarnLine) {
+          ui.writeWarnLine(message);
+        } else {
+          ui.writeLine(message, 'WARN');
+        }
+      },
+
+      error: function(message) {
+        // fallback needed for support of ember-cli < 2.2.0
+        if (ui.writeError) {
+          ui.writeError(message);
+        } else {
+          ui.writeLine(message, 'ERROR');
+        }
+      }
+    };
 
-  // Don't include the `includePolyfill` flag, since Babel doesn't care
-  delete options.includePolyfill;
+    // Ensure modules aren't compiled unless explicitly set to compile
+    options.blacklist = options.blacklist || ['es6.modules'];
 
-  if (options.compileModules === true) {
-    if (options.blacklist.indexOf('es6.modules') >= 0) {
-      options.blacklist.splice(options.blacklist.indexOf('es6.modules'), 1);
+    // do not enable non-standard transforms
+    if (!('nonStandard' in options)) {
+      options.nonStandard = false;
     }
 
+    // Remove custom options from `options` hash that is passed to Babel
+    delete options.includePolyfill;
     delete options.compileModules;
-  } else {
-    if (options.blacklist.indexOf('es6.modules') < 0) {
-      options.blacklist.push('es6.modules');
+
+    if (compileModules) {
+      if (options.blacklist.indexOf('es6.modules') >= 0) {
```

Comment:
again: this existed already before this PR (see https://github.com/babel/ember-cli-babel/blob/b9a0dc232f4038bf6c95de51f71b04faff08406d/index.js#L125-L135)